code/modules/cm_marines/overwatch.dm

```diff
@@ -1065,6 +489,384 @@
+
+		if("set_secondary")
+			if(operator != usr)
+				return FALSE
+			var/input = sanitize_control_chars(sanitize(params["objective"]))
+			current_squad.secondary_objective = input + " ([worldtime2text()])"
+			send_to_squad("Your secondary objective has been changed to '[input]'. See Status pane for details.")
+			send_maptext_to_squad(input, "Secondary Objective Updated:")
+			visible_message("[icon2html(src, viewers(src))] [SPAN_BOLDNOTICE("Secondary objective of squad '[current_squad]' set to '[input]'.")]")
+			log_overwatch("[key_name(usr)] set [current_squad]'s secondary objective to '[input]'.")
+			return TRUE
+
+		if("set_supply")
+			if(operator != usr)
+				return FALSE
+			x_supply = params["target_x"]
+			y_supply = params["target_y"]
+			to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Latitude is now [y_supply], longitude is now [x_supply].")]")
+			return TRUE
+
+		if("set_bomb")
+			if(operator != usr)
+				return FALSE
+			x_bomb = params["target_x"]
+			y_bomb = params["target_y"]
+			to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Latitude is now [y_bomb], longitude is now [x_bomb].")]")
+			return TRUE
+
+		if("hide_dead")
+			if(operator != usr)
+				return FALSE
+			dead_hidden = !dead_hidden
+			if(dead_hidden)
+				to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Dead marines are now not shown.")]")
+			else
+				to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Dead marines are now shown again.")]")
+			return TRUE
+
+		if("choose_z")
+			if(operator != usr)
+				return FALSE
+			switch(params["area_picked"])
+				if("Ship")
+					z_hidden = HIDE_ALMAYER
+					to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Marines on the Almayer are now hidden.")]")
+				if("Ground")
+					z_hidden = HIDE_GROUND
+					to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Marines on the ground are now hidden.")]")
+				if("None")
+					z_hidden = HIDE_NONE
+					to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("No location is ignored anymore.")]")
+			return TRUE
+
+		if("toggle_marine_filter")
+			if(operator != usr)
+				return FALSE
+			if(marine_filter_enabled)
+				marine_filter_enabled = FALSE
+				to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("All marines will now be shown regardless of filter.")]")
+			else
+				marine_filter_enabled = TRUE
+				to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Individual Marine Filter is now enabled.")]")
+			return TRUE
+
+		if("filter_marine")
+			if(operator != usr)
+				return FALSE
+			if(current_squad)
+				var/squaddie = params["squaddie"]
+				if(!(squaddie in marine_filter))
+					marine_filter += squaddie
+					to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Marine now hidden.")]")
+				else
+					marine_filter -= squaddie
+					to_chat(usr, "[icon2html(src, usr)] [SPAN_NOTICE("Marine will now be shown.")]")
+			return TRUE
+
+		if("change_lead")
+			if(operator != usr)
+				return FALSE
+			change_lead(params["marine_picked"])
+			return TRUE
+
+		if("insubordination")
```

Comment:
Is there a way to remove insubordination?

Reply:
as it currently exists, it marks them for arrest on security terminals, which'd require using a security terminal to remove